utils/python/CIME/XML/env_archive.py

```diff
@@ -3,14 +3,27 @@
 """
 from standard_module_setup import *
 
-from env_base import EnvBase
+from CIME.XML.generic_xml import GenericXML
+from CIME.XML.archive import Archive
+from CIME.XML.headers import Headers
 
 logger = logging.getLogger(__name__)
 
-class EnvArchive(EnvBase):
+class EnvArchive(GenericXML):
 
     def __init__(self, case_root=os.getcwd(), infile="env_archive.xml"):
         """
         initialize an object interface to file env_archive.xml in the case directory
         """
-        EnvBase.__init__(self, case_root, infile)
+        if os.path.abspath(infile):
+            fullpath = infile
+        else:
+            fullpath = os.path.join(case_root, infile)
+        GenericXML.__init__(self, fullpath)
+        if not os.path.isfile(infile):
```

Comment:
Currently this file is just a copy of cime_config/<model>/archive.xml with a header node added.